trello_items.py

```diff
@@ -0,0 +1,209 @@
+import os
+import requests
+from item import Item
+
+"""
+A Global to be used to cache the doard ID
+"""
+BOARD_ID = ""
+
+
+def build_url(endpoint):
+    """
+    Builds the API url from a base URL and the specific endpoint for Trello.
+
+    Args:
+        endpoint: The endpoint to be called.
+
+    Returns:
+        url: The full url for the endpoint.
+    """
+    return os.getenv('TRELLO_BASE_URL') + endpoint
+
+
+def get_default_params():
+    """
+    Gets the default parameter to be added to all calls to Trello. Specifically the key 
+    and the token
+
+    Returns:
+        params: A map with the default parameters.
+    """
+    return {
+        'key': os.getenv('TRELLO_KEY'),
+        'token': os.getenv('TRELLO_TOKEN')
+    }
+
+
+def get_boards():
+    """
+    Gets all the boards for the currnt user
+
+    Returns:
+        boards: json object containing the baords.
+    """
+    params = get_default_params()
+    url = build_url('/members/me/boards')
+
+    response = requests.get(url, params=params)
+    boards = response.json()
+
+    return boards
+
+
+def get_board_id():
+    """
+    Gets the board ID which is to be used for this application from Trello id not already cached
+
+    Returns:
+        id: The board ID.
+    """
+    global BOARD_ID
+    if BOARD_ID == "":
+        boards = get_boards()
+        BOARD_ID = next(
+            (board['id'] for board in boards if board['name'] == "ToDoBoard"), None)
+    return BOARD_ID
+
+
+def get_lists(include_cards=True):
+    """
+    Fetches all lists from the board in Trello.
+
+    Args:
+        include_cards: A flag to indicate is the items in the list should also be returned.
+
+    Returns:
+        list: The list of saved items.
+    """
+    board_id = get_board_id()
+
+    params = get_default_params()
+    if include_cards:
+        params.update({'cards': 'all'})
+    url = build_url('/boards/%s/lists' % board_id)
+
+    response = requests.get(url, params=params)
+    lists = response.json()
+
+    return lists
+
+
+def get_list(name):
+    """
+    Fetches the list with the sepcific name.
+
+    Args:
+        name: The name of the list to return.
+
+    Returns:
+        item: The list of items, or None if no list match the specified name.
+    """
+    lists = get_lists(False)
+    return next((list for list in lists if list['name'] == name), None)
+
+
+def get_items():
+    """
+    Fetches all saved items from the Trello.
+
+    Returns:
+        list: The list of saved items.
+    """
+    card_lists = get_lists()
+
+    items = []
+    for card_list in card_lists:
```

Comment:
Probably easiest to flatten this out using a comprehension: https://stackoverflow.com/questions/952914/how-to-make-a-flat-list-out-of-list-of-lists

Reply:
I'll change